### PR TITLE
Add new method Class.isIdentity()

### LINKED ISSUE #16027: valhalla error: cannot find symbol method isIdentity()
```
/Users/ivm/ext-valhalla/1003/testbuilds/src/java.base/share/classes/java/util/Objects.java:202: error: cannot find symbol
        return obj.getClass().isIdentity() ||  // Before Valhalla all classes are identity classes
                             ^
  symbol:   method isIdentity()
  location: class Class<CAP#1>
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Object from capture of ? extends Object
```

The change is in `openj9-staging` branch.

FYI @hangshao0

### PR DESCRIPTION
**Pull request by hangshao0**
1. Add implementation of Class.isIdentity()
2. Update tests

Closes #16027 #16031

Signed-off-by: Hang Shao <hangshao@ca.ibm.com>

### CODE REVIEW

**Comment by hangshao0** [comment]
For potential PR builds, the functional Valhalla tests won't compile at this moment. The compilation errors should be resolved by https://github.com/eclipse-openj9/openj9/pull/16040.

**Comment by hangshao0** [comment]
I've closed https://github.com/eclipse-openj9/openj9/pull/16040 and moved all the changes here.

File: runtime/vm/romclasses.c [hunk 1]
```diff
@@ -127,6 +127,10 @@ initializeArrayROMClass(J9ROMArrayClass *romClass, J9UTF8 *className, U_32 array
 	NNSRP_SET(romClass->className, className);
 	NNSRP_SET(romClass->superclassName, &arrayROMClasses.objectClassName);
 	romClass->modifiers = J9AccFinal | J9AccPublic | J9AccClassArray | J9AccAbstract;
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	/* Arrays are always identity classes. */
+	romClass->modifiers |= J9AccClassHasIdentity;
```
Comment (tajila) [inline runtime/vm/romclasses.c:132]: do we add the flag for non-Arrays as well?

Comment (hangshao0) [reply runtime/vm/romclasses.c:132]: Yes, for non-array classes it is set here:

https://github.com/eclipse-openj9/openj9/blob/52ab41fa82505d2b245129429bd419591b7d17d1/runtime/bcutil/ClassFileOracle.cpp#L817-L819

**Comment by tajila** [comment]
Jenkins test sanity,extended xlinuxval jdknext

**Comment by hangshao0** [comment]
The PR build failure is due to https://github.com/eclipse-openj9/openj9/issues/16028.

**Pull request merged**